libs/vertexai/langchain_google_vertexai/chat_models.py

```diff
@@ -600,7 +600,330 @@ async def _completion_with_retry_inner(
+
+    Tool calling:
+        .. code-block:: python
+
+            from langchain_core.pydantic_v1 import BaseModel, Field
+
+            class GetWeather(BaseModel):
+                '''Get the current weather in a given location'''
+
+                location: str = Field(..., description="The city and state, e.g. San Francisco, CA")
+
+            class GetPopulation(BaseModel):
+                '''Get the current population in a given location'''
+
+                location: str = Field(..., description="The city and state, e.g. San Francisco, CA")
+
+            llm_with_tools = llm.bind_tools([GetWeather, GetPopulation])
+            ai_msg = llm_with_tools.invoke("Which city is hotter today and which is bigger: LA or NY?")
+            ai_msg.tool_calls
+
+        .. code-block:: python
+
+            [{'name': 'GetWeather',
+              'args': {'location': 'Los Angeles, CA'},
+              'id': '2a2401fa-40db-470d-83ce-4e52de910d9e'},
+             {'name': 'GetWeather',
+              'args': {'location': 'New York City, NY'},
+              'id': '96761deb-ab7f-4ef9-b4b4-6d44562fc46e'},
+             {'name': 'GetPopulation',
+              'args': {'location': 'Los Angeles, CA'},
+              'id': '9147d532-abee-43a2-adb5-12f164300484'},
+             {'name': 'GetPopulation',
+              'args': {'location': 'New York City, NY'},
+              'id': 'c43374ea-bde5-49ca-8487-5b83ebeea1e6'}]
+
+        See ``ChatVertexAI.bind_tools()`` method for more.
+
+    Structured output:
+        .. code-block:: python
+
+            from typing import Optional
+
+            from langchain_core.pydantic_v1 import BaseModel, Field
+
+            class Joke(BaseModel):
+                '''Joke to tell user.'''
+
+                setup: str = Field(description="The setup of the joke")
+                punchline: str = Field(description="The punchline to the joke")
+                rating: Optional[int] = Field(description="How funny the joke is, from 1 to 10")
+
+            structured_llm = llm.with_structured_output(Joke)
+            structured_llm.invoke("Tell me a joke about cats")
+
+        .. code-block:: python
+
+            Joke(setup='What do you call a cat that loves to bowl?', punchline='An alley cat!', rating=None)
+
+        See ``ChatVertexAI.with_structured_output()`` for more.
+
+    Image input:
+        .. code-block:: python
+
+            import base64
+            import httpx
+            from langchain_core.messages import HumanMessage
+
+            image_url = "https://upload.wikimedia.org/wikipedia/commons/thumb/d/dd/Gfp-wisconsin-madison-the-nature-boardwalk.jpg/2560px-Gfp-wisconsin-madison-the-nature-boardwalk.jpg"
```

Comment:
nits: maybe add a hint that if an image points to a GCS bucket then it's faster / more efficient since bytes are transferred back and forth